### ISSUE TITLE
Error message when run

### ISSUE THREAD

**Issue by hugokennes** [stas00/google-earth-grid#1]
Dear , like with Tom ancient versions , when i copy Your  new code and run,
i instantly get the message SyntaxError: multiple statements found while compiling a single statement
The run  stops on the first command " import math "  ? i guess i must make an import library , math etc etc ?
And then , how to get the Google Earth Utilities necessary to import ?
I never got an answer of Tom. I made many simulations successfully online with his webpage during 10 years
Then tried to download his sources to work offline.
I have downloaded Python 3.10 today.
Can i contact You on e -mail ? Thx a lot

**Comment by stas00** [comment]
looks like you copy-n-pasted the script instead of properly downloading them and your pasting broke the indentation. I'm only guessing based on the error message you shared, please see: https://stackoverflow.com/questions/21226808/syntaxerror-multiple-statements-found-while-compiling-a-single-statement

If you don't know how to clone git repository, here what you can do instead

1 .go to  https://github.com/stas00/google-earth-grid
2. in upper right corner click on [Code] and choose download zip
3. it will save the whole repository as a zip file which you then extract on your computer 

now you have the whole code repository extracted and you should be able to run these scripts.

You may need to prefix the command with `python` like so:

```
python ./makegrid.py -h
```

you're not tell me which OS you're on. The above would work for linux, may be for Mac, for windows you probably need `python.exe`? not sure.

if the above didn't help, the next step is for you to show exactly what you did, i.e. which exact command did you run and which full error message you received - not just the last line of it.

and of course your environment - windows/linux/mac/etc.

> Can i contact You on e -mail ?

there is no need to - I can answer your questions here and it may help others in the future - if we do it by email others won't benefit from it.

**Comment by hugokennes** [comment]
thx for quick answer , i downloaded & use Python ( unknown for me ) under windows10 , giving the screen and start message 
"Python 3.10.5 (tags/v3.10.5:f377153, Jun  6 2022, 16:14:13) [MSC v.1929 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license()" for more information." , 
after which i can add the
Python commands
Meantime i used the CODE function , had to buy for 40 euro Winrar  , and so see 4 different python scripts mentioned ( i still have to find how to view and  copy their contents after extracting them in Winrar ( unknown for me ) ) : Python files : utilities , bearing , and coordinates under subcatolog PORT and makegrid.py under catalog Google Earth . I guess i have to put the 4 sources together under the Python runner start message ? Or must i run them separately , and save them for the import commands in the main coding ?
I guess i must put the command python ./makegrid.py -h on top of the script of the main code file ? makegrid.py
Meantime i try to view the expanded scripts and copy & paste them
Since 10 years i have several FBgroups about those Google Earth simulations ( 30.000 members together ) , the KML files generated with Montalk webpage can even be downloaded by the members from the FB file sections and opened in Google Earth. In fact we are Google Earth user groups for the grids related , showing their usage in ancient socalled leylines.
I m 77 old and try to save that Google overlay generator for later generations & FB members in Python 3 and Windows 10 - 11 and offer the possibility to use it also off - line. I always informed retired Prof. Hagens about usage , findings and new applications , she will be delighted it may now eventualy run under the latest windows & python software. 
KML FB files examples https://www.facebook.com/groups/791622274195042/files in several subgroups also available examples https://www.facebook.com/groups/1295386420476634/files Kindest regards

**Comment by hugokennes** [comment]
Additional details asked : example of error message : start coding runned was : 
Python 3.10.5 (tags/v3.10.5:f377153, Jun  6 2022, 16:14:13) [MSC v.1929 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license()" for more information.
#!/usr/bin/env python3


##================= PhiGrid Platonic Coordinate Generator =======================
##
## Summary: This script generates a list of coordinates for the vertices
## of platonic solids, formatted to be pasted directly into the PhiGrid
## script editor.
##
## Instructions:
## 1) Type the platonic shape
## 2) Enter the locking coordinate - what lat/lon a vertex should lock onto
## 3) Enter the alignment coordinate - toward what lat/lon the vertex otherwise 
##                                     directly north of the locking coordinate
##                                     should be rotated.
## 4) Paste the output (starting with PAB) into the PhiGrid script editor, run it.
##
##Query:       Inputs:
##------       ---------
##
##Shape?       tetrahedron, octahedron, cube, icosahedron, dodecahedron, beckerhagens
##
##Latitude?    # (for North), -# (for South)
##
##Longitude?   # (for East), -# (for West)
##
##Example:
##--------
##
##Shape? dodecahedron
##Latitude of lock coordinate? 5
##Longitude of lock coordinate? 5
##Latitude of header coordinate? 30
##Longitude of header coordinate? 31
##
##This will place one corner of the tetrahedron at 5 N 5 E and rotate its edge
##toward 30N 31E -- toward the Great Pyramids at Ghiza.
##
##===============================================================================

import math
.
.
.
ON IMPORT ( math ) LINE (still like always ) immediately reaction : and error text below at end of copied coding : 
.
.
.
dodecahedron, beckerhagens')
        shape = input('\nWhich Shape? ')

        # get lock coordinate
        lat1=input('Latitude of lock coordinate? ')
        lon1=input('Longitude of lock coordinate? ')
        lat2=input('Latitude of header coordinate? ')
        lon2=input('Longitude of header coordinate? ')
        
    run(shape, lat1, lon1, lat2, lon2, outfile)
    
SyntaxError: multiple statements found while compiling a single statement ( coding line is highlighted RED after "import math" )

If i "prefix the command with python like so:

python ./makegrid.py -h" ( like suggested Your comment )  i get immediately "wrong syntax" at the end of codings and the "./" is highlighted in RED in the runned script

( More general comments are  in previous comment before ) Thx for looking & helping

**Comment by stas00** [comment]
the `"./"` is a unix way of telling this file is relative to the current directory (folder). since now we know you're on windows, you most likely need something like:

```
python.exe makegrid.py -h
```

I haven't tried running it on windows so I have no idea if it works or not there.

-----------------
```
SyntaxError: multiple statements found while compiling a single statement ( coding line is highlighted RED after "import math" )
```

as I mentioned earlier searching the internet for this error - it typically means that something gets messed up in the file transfer and python fails to parse this file.

It might be related to the end of line character - windows and linux have different line endings, `\r\n` and `\n` correspondingly. e.g. see: https://stackoverflow.com/questions/426397/do-line-endings-differ-between-windows-and-linux 

Usually one can run `unix2dos` to convert the ends of line to the one matching windows. a quick search gave this site https://sourceforge.net/projects/dos2unix/ to download this free utility from. I have no idea if it's a safe download.

Do you by chance have `perl` installed? If yes I can tell you how to convert the files using perl easily and requiring no additional software. If not,  `unix2dos` should do the trick.

Bottom line - my understanding is that you're not having an issue with this particular grid making software but with getting any python program to run saved from the internet - so need to figure that one first and then I trust this program will run as well.

**Comment by hugokennes** [comment]
Dear , thx , tests :  "python.exe makegrid.py -h" :  m is marked in  red as "syntax error" ; further :  (first command) "import math \r\n" : r is marked red " SyntaxError: unexpected character after line continuation character " ; idem \n ; but :  import math \
import sys , i of next import command is marked red "SyntaxError: invalid syntax" , the first import command is however accepted ?! ; don't have perl , will try dos2unix , clearly a line continuation problem somehow
General : indeed since 10 year we very succesfully used the "online" webpage Montalk Google Earth grids simulator - KML generator https://montalk.net/coordinates.htm But Tom also from the beginning offered for working offline a download of his Python 2 ancient version , i informed him years ago about the same problems as today ( sources unaccepted). Now he mentioned Your conversion Python 3 on his page https://montalk.net/science/115/earth-grid-research , so i tried again giving still exactly the same problems as 10 years ago ! Will try dos2unix , thx for suggestion !

**Comment by hugokennes** [comment]
Dear , 1 comment above , further , downloading Dos2Unix  , the programfile Unix2dos.exe was immediately blocked & deleted by Norton security application , after extraction under Winrar. Its a pitty. If You have eventualy the script under Dos with the correct end of lines embedded , my e-mail for eventual forwarding is  hugo.kennes@gmail.com. Thx a lot any way. ( I know my gran son 16yr in industrial schooling receives second year python programming from his teacher informatics , in july - august holidays i will ask eventual help ) Kindest regards

**Comment by hugokennes** [comment]
Dear , 2 comment above , finally i have to type in all myself by pushing the CR/LF button myself : 

Python 3.10.5 (tags/v3.10.5:f377153, Jun  6 2022, 16:14:13) [MSC v.1929 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license()" for more information.
( command )
from google_earth_utils import rotateX, rotateY, rotateZ, bearing, coordinates, get_shape
(run example )
( error message ) : 
Traceback (most recent call last):
  File "<pyshell#0>", line 1, in <module>
    from google_earth_utils import rotateX, rotateY, rotateZ, bearing, coordinates, get_shape
ModuleNotFoundError: No module named 'google_earth_utils'

( Since i only need coordinates function Google Earth Montalk app , i guess i can integrate the code
related google earth utils modules in the same script avoiding the necessary import )

Typing in this way the whole 2 codings is non-sense, when i can use the webpage of Montalk successfully
I will inform my FB "earthgridders - members" about Your existing Unix conversion in Python 3 for those
people using Unix

Thx so much for advises & help in Your precious time !

**Comment by stas00** [comment]
So it's pretty clear that there is no problem with the program, but you just need to get someone who knows how to run python on windows. surely, typing it out by hand is not going to work. Perhaps find a forum that can support you.

But since you're hands on, you could try reading the instructional articles, e.g.:
https://www.google.com/search?client=firefox-b-d&q=running+python+on+windows

so I'd leave this repo alone for some time and figure out how to run python programs on windows (including how to deal with different line endings which clearly break python.

Once you understand the nuances of running python on windows then using this repo should be easy.

Apologies I can't help you any further since I don't know anything about windows. I'm a linux user.

re: `unix2dos` - you can do it in python as well, but it's a bit of a chicken-n-egg problem if you can't run python programs, but have a look https://www.google.com/search?q=%22unix2dos%22+in+python perhaps you will be able to find a solution in one of the results there.

Good luck and I'd be happy to support you once you get the python on windows sorted out.
